pyanaconda/ui/gui/spokes/custom_storage.py

```diff
@@ -1569,22 +1569,28 @@ def on_raid_level_changed(self, widget):
         self.on_value_changed()
 
     def on_size_changed(self, widget):
+        """Callback for text change in "desired capacity" widget"""
         if not self._sizeEntry.get_sensitive():
             return
 
-        current_size = Size(self._request.device_size)
-        displayed_size = current_size.human_readable(max_places=self.MAX_SIZE_PLACES)
-
-        if displayed_size == self._sizeEntry.get_text():
-            return
-
         size = get_size_from_entry(
             self._sizeEntry,
             lower_bound=self.MIN_SIZE_ENTRY,
             units=SIZE_UNITS_DEFAULT
         )
 
         if size is None:
+            self.set_warning(_(
+              "Values for Desired Capacity can contain only numbers, decimal separator, and unit."
+            ))
+            return
+        else:
+            self.clear_info()
```

Comment:
I think that this should be called before `set_warning` as well.

Reply:
That would animate the bar at every keystroke :) I changed it a bit though, so it clears only if the warning is the same.

It might be better to actually validate the value only on applying with `on_configure_clicked`, but I think this requires less change to the overall structure.

Reply:
Good point. I think that is the reason why we use `show_warning_message` for password entries.